lib/dor/rights_auth.rb

```diff
@@ -423,12 +423,12 @@ def self.init_index_elements(doc)
     # rights that make more sense at the object level (e.g. 'dark').
     def self.primary_access_rights(index_terms, errors)
       has_rule = index_terms.include? 'has_rule'
-      if index_terms.include?('none_discover')
+      if index_terms.include?('none_discover') ||
+         errors.include?('no_discover_access') ||
+         errors.include?('no_discover_machine')
```

Comment:
i can't tell whether the branches that were consolidated in this PR's change were intended to be kept separate.  that is, if `index_terms` includes `cdl_none` as well as `no_read_machine` or `none_read`, should the `cdl_none` take precedence, leading this method to produce a result of `'controlled digital lending'` instead of `'dark'`?  or was that just an artifact of sticking the `'cdl_none'` check into an already bifurcated conditional (at the time, definitely unnecessarily bifurcated, since the conditionals were right next to each other).

maybe @peetucket can weigh in, since he added the `'cdl_none'` check.  it's been a while since i've worked in dor-rights-auth, so i'm a little fuzzy on it's complex wrangling of the `rightsMetadata` datastream.

if it turns out the bifurcation is necessary (i suspect it isn't, but 🤷), we should probably add a test that this PR would break.

Reply:
the commit that added the `cdl_none` check:  https://github.com/sul-dlss/dor-rights-auth/commit/342327c74a0a16835dace8c619167968f8892c8e#diff-0a1ca0c5e439f17160515030dfbb57f9209917dd356320f48127bb5345dccc50R421-R422

and its PR:  https://github.com/sul-dlss/dor-rights-auth/pull/45/files?file-filters%5B%5D=.rb#diff-0a1ca0c5e439f17160515030dfbb57f9209917dd356320f48127bb5345dccc50R422-R423

Reply:
Good call, @jmartin-sul. I'll defer to @peetucket and I'm happy to make changes if needed.

Reply:
I think this is fine.  I think in the real world, `dark` and `cdl_none` are going to be mutually exclusive, so order of precedence in this method shouldn't matter.

Reply:
thanks @peetucket!